lib/auth/tls_test.go

```diff
@@ -3940,15 +3989,19 @@ func TestGRPCServer_CreateTokenV2(t *testing.T) {
 			err = client.CreateToken(ctx, tt.token)
 			tt.requireError(t, err)
 
-			if tt.requireEventEmitted {
-				lastEvent := mockEmitter.LastEvent()
-				require.NotNil(t, lastEvent)
-				require.Equal(
-					t,
-					events.TrustedClusterTokenCreateEvent,
-					lastEvent.GetType(),
-				)
+			// Expect last emitted audit events to match the expected ones.
+			emittedEvents := mockEmitter.Events()
+			emittedEventsSliceStart := len(emittedEvents) - len(tt.auditEvents)
+			if emittedEventsSliceStart < 0 {
+				emittedEventsSliceStart = 0
 			}
+			require.Empty(t, cmp.Diff(
+				tt.auditEvents,
+				emittedEvents[emittedEventsSliceStart:],
```

Comment:
> "assert this list contains elements that match given predicate in any order",

Doesn't `require.ElementsMatch` do exactly this?

Reply:
@zmb3 No, it's about a subset match, not matching _all_ contents. Yes, I could have just matched all contents, but I thought it would make the test brittle. A brittle test is bad, because it puts a burden on maintenance when an unrelated feature is being added. But again, this might be "just the way it's done" in Go.

Reply:
I don't think it has anything to do with Go. Our assertions come from a [3rd party package](https://pkg.go.dev/github.com/stretchr/testify/assert).

```
// test that elements match in any order
require.ElementsMatch(t, a, b)

// test subset
require.Contains(t, list, "a")
require.Contains(t, list, "b")
require.Contains(t, list, "c")
```

What else is missing?

Reply:
`require.ElementsMatch()` doesn't ignores order, but doesn't test for a _subset_. I can express "require that all events match, ignoring order", but I can't say "require that _some_ elements match".
`require.Contains()` would solve the issue, but just as `require.ElementsMatch()`, it doesn't support _partial matching_.

I would need a combination of both. Something that could be done in Jest in a way that is described here: https://medium.com/@andrei.pfeiffer/jest-matching-objects-in-array-50fe2f4d6b98. (You can ignore the "Adding it as a custom matcher" section, the piece of code right above it should be self-explanatory.) Our assertion libraries don't contain a tool to do exactly that — unless there's some way of using `go-cmp`'s diffing options in a way I don't see, in which case please let me know. It may come in handy (I don't know, a custom comparator? A transformer from a slice to map?).

I think that the closest thing that I could do to an ideal state would be to `assert.True(t, slices.ContainFunc(...))`, and to compensate for a potentially useless error message (we trade off the verbosity of `cmp.Diff()`), I can just give a custom error message as the third parameter. I'll change it and see what it looks like.